dm/dm/worker/subtask.go

```diff
@@ -552,12 +552,17 @@ func (st *SubTask) markResultCanceled() bool {
 func (st *SubTask) Result() *pb.ProcessResult {
 	st.RLock()
 	defer st.RUnlock()
-	tempProcessResult, err := st.result.Marshal()
-	if err != nil {
-		log.L().Panic("Marshal of the process result is not successful.")
-	}
-	newProcessResult := &pb.ProcessResult{}
-	_ = newProcessResult.Unmarshal(tempProcessResult)
+	tempProcessResult := st.result
+	Errors := []*pb.ProcessError{}
+	Detail := []byte{}
+	immProcessResult := pb.ProcessResult{
+		IsCanceled: false,
+		Errors:     Errors,
+		Detail:     Detail,
+	}
+	var newProcessResult *pb.ProcessResult
+	newProcessResult = &immProcessResult
+	*newProcessResult = *tempProcessResult
```

Comment:
The `pb.ProcessResult.Errors` is an array of pointers, which should be deep copied as well.
```suggestion
	*newProcessResult = *st.Result
```

Reply:
Modified back to Marshal() and Unmarshal(), which is less efficient than deep copy but easier to maintain (have been discussed).